x-pack/legacy/plugins/siem/server/lib/timeline/routes/schemas/export_timelines_schema.ts

```diff
@@ -4,17 +4,41 @@
  * you may not use this file except in compliance with the Elastic License.
  */
 
-import Joi from 'joi';
+import { fold } from 'fp-ts/lib/Either';
+import { pipe } from 'fp-ts/lib/pipeable';
+import * as rt from 'io-ts';
+import { failure } from 'io-ts/lib/PathReporter';
 
-/* eslint-disable @typescript-eslint/camelcase */
-import { ids, exclude_export_details, file_name } from './schemas';
-/* eslint-disable @typescript-eslint/camelcase */
+import {
+  RouteValidationFunction,
+  RouteValidationResultFactory,
+} from '../../../../../../../../../src/core/server';
+import { RequestValidationResult } from '../../types';
 
-export const exportTimelinesSchema = Joi.object({
-  ids,
-}).min(1);
+const ids = rt.array(rt.string);
+export const exportTimelinesSchema = rt.type({ ids });
+
+export const exportTimelinesQuerySchema = rt.type({
+  file_name: rt.string,
+  exclude_export_details: rt.union([rt.literal('true'), rt.literal('false')]),
+});
 
-export const exportTimelinesQuerySchema = Joi.object({
-  file_name: file_name.default('export.ndjson'),
-  exclude_export_details: exclude_export_details.default(false),
+export const exportTimelinesRequestBodySchema = rt.type({
+  ids,
 });
+
+export const decodeOrThrow = <A, O, I>(runtimeType: rt.Type<A, O, I>) => (
+  inputValue: I,
+  validationResult: RouteValidationResultFactory
+) =>
+  pipe(
+    runtimeType.decode(inputValue),
+    fold<rt.Errors, A, RequestValidationResult<A>>(
+      (errors: rt.Errors) => validationResult.badRequest(failure(errors).join('\n')),
+      (validatedInput: A) => validationResult.ok(validatedInput)
+    )
+  );
+
+export const buildRouteValidation = <A, O, I>(
```

Comment:
It would be nice to move these higher up (ideally into `common/`) if possible.

Reply:
Good idea, but since that I haven't seen client side using this, so put it in x-pack/legacy/plugins/siem/server/utils/build_validation